src/makeInfinityAdder.js

```diff
@@ -4,7 +4,23 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (number) => {
+    if (number === undefined) {
+      const result = sum;
+
+      sum = 0;
+
+      return result;
+    }
+
+    sum += number;
```

Comment:
Ensure that the input `number` is indeed a number before adding it to the sum. It's a good practice to validate the input to avoid unexpected behavior. You could use `typeof number === 'number'` for this check.